### PR TITLE
strutil: Add MakeRandomString from pebble

### PR DESCRIPTION
**Pull request by anpep**

### CODE REVIEW

File: strutil/strutil.go [hunk 1]
```diff
@@ -22,13 +22,35 @@ package strutil
...
+func init() {
+	// golang does not init Seed() itself
+	rand.Seed(time.Now().UTC().UnixNano())
+}
+
+const letters = "BCDFGHJKLMNPQRSTVWXYbcdfghjklmnpqrstvwxy0123456789"
+
+// MakeRandomString returns a random string of length length
+//
+// The vowels are omitted to avoid that words are created by pure
```
Comment (flotter) [inline strutil/strutil.go:43]: I was staring at the 'letters' in order to understand why some are missing, and only found the explanation here. Would it perhaps be better to move this up to the const declaration?

File: strutil/strutil.go [hunk 2]
```diff
@@ -22,13 +22,35 @@ package strutil
...
 	"sort"
 	"strconv"
 	"strings"
+	"time"
 	"unicode"
 	"unicode/utf8"
 )
 
+func init() {
+	// golang does not init Seed() itself
```
Comment (flotter) [inline strutil/strutil.go:35]: I would personally not see this as a surprise. (as in this is normal for me to have to do this in order to be in control of how random the pseudo generator appear.

**Review by flotter — APPROVED** [review]
Only unimportant subjective comments :-)

**Review by mvo5 — COMMENTED** [review]
Fwiw, we used to have this helper too but then moved it into `randutil`, see https://github.com/snapcore/snapd/blob/master/randutil/rand.go#L67

The relevant PR for snapd is here https://github.com/snapcore/snapd/pull/8084

Among other things the seeding of the RNG is more elaborate in there. I think we should get the input of Samuele if we want to peruse this PR. The alternative might be to just import the `randutil` change from snapd (maybe even just the string part without the crypto bits).

File: strutil/strutil.go (hunk 2 above)
Comment (mvo5) [reply strutil/strutil.go:35]: (a bit of a pedantic/nitpicking comment, sorry!) What I like about `randutil` (see my other comment) is that it's less surprising to see something messing with the RNG seed on import when it's called `randutil` than when it's called `strutil`.

**Comment by anpep** [comment]
> Fwiw, we used to have this helper too but then moved it into `randutil`, see https://github.com/snapcore/snapd/blob/master/randutil/rand.go#L67
> 
> The relevant PR for snapd is here [snapcore/snapd#8084](https://github.com/snapcore/snapd/pull/8084)
> 
> Among other things the seeding of the RNG is more elaborate in there. I think we should get the input of Samuele if we want to peruse this PR. The alternative might be to just import the `randutil` change from snapd (maybe even just the string part without the crypto bits).

Yes, I think it's better to just port the relevant bits from `randutil`. Thanks for your feedback!

File: strutil/strutil.go (hunk 2 above)
Comment (benhoyt) [reply strutil/strutil.go:35]: I don't think we should ever mess with something like that just from importing. This should be an app-level concern. Just importing something like this shouldn't have a non-obvious side effect.

**Pull request closed without merging**